test/powershell/engine/ExperimentalFeature/ExperimentalFeature.Basic.Tests.ps1

```diff
@@ -0,0 +1,372 @@
+            $originalDefaultParameterValues = $PSDefaultParameterValues.Clone()
+            $PSDefaultParameterValues["it:skip"] = $true
+        } else {
+            $TestModule = Join-Path $PSScriptRoot "assets" "ExpTest"
+            $AssemblyPath = Join-Path $TestModule "ExpTest.dll"
+            if (-not (Test-Path $AssemblyPath)) {
+                ## When using $SourcePath directly, 'Add-Type' fails in AppVeyor CI runs with an 'access denied' error.
+                ## It turns out Pester doesn't handle an exception like this from 'BeforeAll'. It causes the Pester to
+                ## be somehow corrupted, and results in random failures in other tests.
+                ## To work around this issue, we copy the source file to 'TestDrive' before calling 'Add-Type'.
+                $SourcePath = Join-Path $TestModule "ExpTest.cs"
+                $SourcePath = (Copy-Item $SourcePath TestDrive:\ -PassThru).FullName
+                Add-Type -Path $SourcePath -OutputType Library -OutputAssembly $AssemblyPath
+            }
+            $moduleInfo = Import-Module $TestModule -PassThru
+        }
+    }
+
+    AfterAll {
+        if ($skipTest) {
+            $global:PSDefaultParameterValues = $originalDefaultParameterValues
+        } else {
+            Remove-Module -ModuleInfo $moduleInfo -Force -ErrorAction SilentlyContinue
+        }
+    }
+
+    It "No experimental feature is enabled" {
+        $EnabledExperimentalFeatures.Count | Should -Be 0
+    }
+
+    It "Replace existing command <Name> - version one should be shown" -TestCases @(
+        @{ Name = "Invoke-AzureFunction"; CommandType = "Function" }
+        @{ Name = "Invoke-AzureFunctionCSharp"; CommandType = "Cmdlet" }
+    ) {
+        param($Name, $CommandType)
+        $command = Get-Command $Name
+        $command.CommandType | Should -Be $CommandType
+        $command.Source | Should -BeExactly $moduleInfo.Name
+        & $Name -Token "Token" -Command "Command" | Should -BeExactly "Invoke-AzureFunction Version ONE"
+
+        if ($CommandType -eq "Function") {
+            $expectedErrorId = "CommandNotFoundException,Microsoft.PowerShell.Commands.GetCommandCommand"
+            { Get-Command "Invoke-AzureFunctionV2" -ErrorAction Stop } | Should -Throw -ErrorId $expectedErrorId
+            { & $moduleInfo { Get-Command "Invoke-AzureFunctionV2" -ErrorAction Stop } } | Should -Throw -ErrorId $expectedErrorId
+        }
+    }
+
+    It "Experimental parameter set - '<Name>' should NOT have '-SwitchOne' and '-SwitchTwo'" -TestCases @(
+        @{ Name = "Get-GreetingMessage"; CommandType = "Function" }
+        @{ Name = "Get-GreetingMessageCSharp"; CommandType = "Cmdlet" }
+    ) {
+        param($Name, $CommandType)
+        $command = Get-Command $Name
+        $command.CommandType | Should -Be $CommandType
+        ## 11 common parameters + '-Name'
```

Comment:
For future maintenance of this code, it would be better to have something like:

```powershell
BeforeAll {
  $commonParameters = @('a','b','c',...)
}
...
$command.Parameters.Count | Should -Be ($commonParameters + "-Name").Length
```

Apply in all cases below.  I think it removes the need for the comments.

Reply:
I should have used `[System.Management.Automation.Internal.CommonParameters].GetProperties().Length` to get the count of common parameters. I will make the change.